api/handlers/changes.js

```diff
@@ -224,12 +224,21 @@ const getChanges = feed => {
           });
           return;
         }
-        const changes = mergeChangesResponses(responses);
-        if (!changes) {
+        const malformedResponse = responses.find(response => {
+          if (!response || !response.results) {
+            // See: https://github.com/medic/medic-webapp/issues/3099
+            // This should never happen, but apparently it does sometimes.
+            // Attempting to log out the response usefully to see what's occuring
+            console.error('No _changes error, but malformed response:', JSON.stringify(changes));
+            return true;
+          }
+        });
+        if (malformedResponse) {
           return errorResponse(feed, 'No _changes error, but malformed response.');
         }
-        cleanUp(feed);
+        const changes = responses.length === 1 ? responses[0] : mergeChangesResponses(responses);
```

Comment:
This looks like the only line that relates to the PR description...

Reply:
Right, but I had to pull out the malformed handling code because otherwise it wouldn't happen with a single response.